pages/api/session.js

```diff
@@ -1,92 +1,100 @@
+    if (password === undefined) {
+      const verifyWardCode = container.getVerifyWardCode();
+      verifyWardCodeResponse = await verifyWardCode(code);
+    }
 
-      let token = undefined;
-      const tokens = container.getTokenProvider();
-      if (verifyTrustAdminCodeResponse.validTrustAdminCode) {
-        token = tokens.generate({
-          wardId: undefined,
-          wardCode: undefined,
-          trustId: verifyTrustAdminCodeResponse.trust.id,
-          type: TRUST_ADMIN,
-        });
-      } else if (
-        verifyWardCodeResponse.validWardCode &&
-        !verifyTrustAdminCodeResponse.validTrustAdminCode &&
-        !verifyAdminCodeResponse.validAdminCode
-      ) {
-        const { ward } = verifyWardCodeResponse;
-        token = tokens.generate({
-          wardId: ward.id,
-          wardCode: ward.code,
-          trustId: ward.trustId,
-          type: WARD_STAFF,
-        });
-      } else if (verifyAdminCodeResponse.validAdminCode) {
-        token = tokens.generate({
-          wardId: undefined,
-          wardCode: undefined,
-          trustId: undefined,
-          type: ADMIN,
-        });
-      }
+    const verifyTrustAdminCode = container.getVerifyTrustAdminCode();
+    const verifyTrustAdminCodeResponse = await verifyTrustAdminCode(
+      code,
+      password
+    );
 
-      const expiryHours = 14;
-      let expiry = new Date();
-      if (
-        verifyTrustAdminCodeResponse.validTrustAdminCode ||
-        verifyAdminCodeResponse.validAdminCode
-      ) {
-        expiry.setTime(expiry.getTime() + 1 * 60 * 60 * 1000);
-      } else {
-        expiry.setTime(expiry.getTime() + expiryHours * 60 * 60 * 1000);
-      }
+    const verifyAdminCode = container.getVerifyAdminCode();
+    const verifyAdminCodeResponse = await verifyAdminCode(code, password);
 
-      res.writeHead(201, {
-        "Set-Cookie": `token=${token}; httpOnly; path=/; expires=${expiry}`,
-      });
-      res.end(JSON.stringify({ wardId: code }));
+    if (
+      !verifyTrustAdminCodeResponse.validTrustAdminCode &&
+      !verifyWardCodeResponse.validWardCode &&
+      !verifyAdminCodeResponse.validAdminCode
+    ) {
+      res.statusCode = 401;
+      res.end();
       return;
-    } else if (method === "DELETE") {
-      const expired = new Date(0);
+    }
 
-      res.writeHead(201, {
-        "Set-Cookie": `token=''; httpOnly; path=/; expires=${expired}`,
+    let token = undefined;
+    const tokens = container.getTokenProvider();
+    if (verifyTrustAdminCodeResponse.validTrustAdminCode) {
+      token = tokens.generate({
+        wardId: undefined,
+        wardCode: undefined,
+        trustId: verifyTrustAdminCodeResponse.trust.id,
+        type: TRUST_ADMIN,
       });
-      res.end(JSON.stringify({ wardId: code }));
-      return;
+    } else if (
+      verifyWardCodeResponse.validWardCode &&
+      !verifyTrustAdminCodeResponse.validTrustAdminCode &&
+      !verifyAdminCodeResponse.validAdminCode
```

Comment:
Instead of checking all these you can simply move this to an `else if` at the end of the chain because the other conditions will be implicitly false

Reply:
nice!